sds/src/proximity_keywords.rs

```diff
@@ -182,17 +237,14 @@ fn calculate_keyword_pattern(keyword: &str) -> Ast {
         keyword_pattern.push(word_boundary())
     }
 
-    for character in keyword.chars() {
-        let kind = if regex_syntax::is_meta_character(character) {
-            LiteralKind::Meta
-        } else {
-            LiteralKind::Verbatim
-        };
-        keyword_pattern.push(Ast::Literal(Literal {
-            span: span(),
-            kind,
-            c: character,
-        }))
+    for (i, c) in keyword.chars().enumerate() {
+        if i != 0 {
```

Comment:
It's because of how we are matching with the regex. We are just checking if the regex matches _anywhere_ in the prefix. Even without the "allowed chars" added to the front, it will still get a match on the string `_hello_world_`, the match just won't contain the first character, which is fine.

Reply:
ok it makes sense now, it might be blurry if we start to use alphanumeric characters in the list of excluded chars (but I don't think it makes sense), LGTM then